server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -408,6 +408,10 @@ public long getPartitionDataSize(String appId, int shuffleId, int partitionId) {
 
   public long requireBuffer(
       String appId, int shuffleId, List<Integer> partitionIds, int requireSize) {
+    boolean isRegistered = shuffleBufferManager.checkIfRegistered(appId, shuffleId, partitionIds);
+    if (!isRegistered) {
+      return -4; // Keep the same with StatusCode
```

Comment:
Could give `-1` and `-4` a error code? like
```
enum ERRORCODE {
NO_BUFFER(-1),
NO_REGISTER(-4)
}
```
Or we can reuse the error code which existed.

Reply:
I find there is no error code existed I can use, and I craete RequireBufferStatusCode